contracts/LeverageEngine.sol

```diff
@@ -87,4 +102,9 @@ contract LeverageEngine is AccessController {
         _positionToken.burn(positionTokenId);
         _coordinator.unwindLeveragedOUSD(positionTokenId, msg.sender);
     }
+
+    // required - the caller must have allowance for accounts's tokens of at least amount.
+    function _burnArchTokenForPosition(address sender, uint256 archAmount) internal expectInitialized {
```

Comment:
do we need to add a require here? or will it automatically check / revert if not enough

Reply:
automatically revert as the burn method checks allowance

Reply:
perfect